### PR TITLE
fix(mobile): mount the swipe target's neighbor while the day snap animates

### PR DESCRIPTION
**Pull request by maccman**
## Problem

Two follow-ups to #576, which fixed the day-swipe snap animation by deferring everything a swipe triggers to Embla's `settle` event. That deferral swept up two things it shouldn't have:

1. **`selectedIndex`**, which the carousel's mount radius follows — so the slide beyond the swipe target only mounted after the animation finished, and a quick double swipe dragged onto a still-empty spacer (blank note until the animation settled). Single swipes were unaffected.
2. **The calendar strip**, which follows the route `date` — so the strip's selection highlight and the rolling month title (#585) waited out the snap animation instead of moving with the gesture.

## Change

Split the swipe's consequences across the two Embla events, in `use-day-carousel.ts`:

- **`select` (pointer-up):** two light things follow the swipe target immediately, wrapped in one `startTransition` so React fits the work around the animation's frames instead of blocking the release frame (Embla's fixed-timestep loop has no lag cap — a long stall there fast-forwards the animation, the original #576 bug):
  - `selectedIndex` — the incoming neighbor's `DaySlide` mounts while the animation plays, so a second swipe lands on real content;
  - a new `onTarget` callback announcing the destination day — the daily screen holds it as `targetDate` and passes `targetDate ?? date` to the calendar strip, so the month title rolls and the strip highlight moves at pointer-up, ahead of the route. The override expires whenever the route `date` moves (render-phase previous-value pattern), so navigations that bypass `select` — daily backlinks, history — supersede it too (Bugbot caught the stale-override case).
- **`settle`:** unchanged — the day report (route navigation, full-surface re-render) and the edge re-center stay deferred until the animation is done.

Also adds hook-level tests for the swipe wiring, driven through a fake Embla API (the harness pattern from the week-strip tests): target-follow and `onTarget` at `select` without reporting, report-on-`settle`, the route-echo guard, external in-window jumps, and edge re-centering.

## Verification

- `pnpm --filter @reflect/desktop test --run src/mobile/use-day-carousel.test.ts src/mobile/mobile-screen.test.tsx` — 41 passed (6 new)
- `pnpm check` (typecheck + lint) — clean
- Not verified on-device — the swipe feel needs the owed device pass: quick double-swipe (no blank slide), cross-month swipe (title rolls during the gesture), and single-swipe smoothness. Worst case a heavy editor mount mid-animation costs a frame or two, which is the deliberate trade against blank content.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

> [!NOTE]
> **Medium Risk**
> Touches mobile daily navigation and carousel/route sync; behavior is localized and covered by new hook tests, but swipe feel and double-swipe edge cases still need on-device validation.
> 
> **Overview**
> After deferring all swipe work to Embla `settle` (#576), the calendar strip and `selectedIndex` (slide mount radius) lagged the gesture—causing blank slides on quick double-swipes and a delayed month title.
> 
> **`useDayCarousel`** now listens to **`select`** (pointer-up) for lightweight updates inside `startTransition`: bump `selectedIndex` so the target neighbor’s editor mounts during the snap, and call a new **`onTarget`** with the destination day. **Route navigation** via **`onSelect`** still runs only on **`settle`**, preserving smooth snap animation.
> 
> **`MobileDaily`** holds `targetDate` from `onTarget` and feeds **`CalendarStrip`** `targetDate ?? date` so highlight and month title move with the swipe; any real `date` change or explicit `select` clears the override.
> 
> Hook tests use a fake Embla API to assert `select` vs `settle` behavior, echo guard, and edge re-centering.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit f954a3aa2fea8750dbb5d313ac50849933e28bea. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: apps/desktop/src/mobile/screens/daily.tsx [hunk 1]
```diff
@@ -26,9 +27,16 @@ export function MobileDaily({ date }: { date: string }): ReactElement {
...
+  // animation plays, instead of waiting for the settle-time route change.
+  // Any actual selection clears it: by then the route is the truth again.
+  const [targetDate, setTargetDate] = useState<string | null>(null)
   const select = useCallback(
-    (day: string): void =>
-      navigate(day === today ? { kind: 'today' } : { kind: 'daily', date: day }),
+    (day: string): void => {
+      setTargetDate(null)
+      navigate(day === today ? { kind: 'today' } : { kind: 'daily', date: day })
+    },
```
[the remark this answers is not in the document]
Comment (maccman) [reply apps/desktop/src/mobile/screens/daily.tsx:48]: Fixed in f954a3aa: the override now expires on any `date` change (render-phase previous-value pattern in `MobileDaily`), so navigations that bypass `select` — daily backlinks, history — supersede it too. The `select`-time clear remains for same-date taps.
